### PR TITLE
Resnet data loading

### PR DESCRIPTION
**Pull request by phelps-matthew**
Incorporate better dataloaders from https://gist.github.com/kevinzakka/d33bf8d6c7f06a9d8c76d97a7879f5cb. 
New developments:
- image sampling
- validation dataloader (main motivation of pull request)
- normalizes dataloader images according to official pytorch CIFAR tutorial
- updates argparse arguments accordingly, with a few new additions (data_dir, log_dir)
- consolidates dataloading into separate module
- adds
Thanks!

### CODE REVIEW

**Comment by phelps-matthew** [comment]
Looks good, thank you Matthew!
